### PR TITLE
script for driving light controllers

### PR DESCRIPTION
**Pull request by markusb**
Here the script for driving light controllers.

3 files: xml, js and README.

Have tested it on Ubuntu 14.04 LTS 64bit with Mixxx 1.11.0.

### CODE REVIEW

**Comment by markusb** [comment]
I went on to test the script with the latest development release from git (1.12.0-alpha (build master r5169).

```
It does not work. I can not even see the 'print' messages in the log/stdout I was in 1.11. Not even the print messages in the 'ini' function. I do see the messages associated with loading the script, so it detects my connections, but it does not call them !

Debug [Controller]: ControllerEngine: Loading & evaluating all script code 
Debug [Controller]: ControllerEngine: Watching JS File: "/home/markus/Workspace/Mixxx/res/controllers/common-controller-scripts.js" 
Debug [Controller]: ControllerEngine: Loading "/home/markus/Workspace/Mixxx/res/controllers/common-controller-scripts.js" 
Debug [Controller]: ControllerEngine: Watching JS File: "/home/markus/Workspace/Mixxx/mixxx-settings/controllers/MIDI_for_LightEffects.js" 
Debug [Controller]: ControllerEngine: Loading "/home/markus/Workspace/Mixxx/mixxx-settings/controllers/MIDI_for_LightEffects.js" 
Warning [Controller]: ControllerEngine: No "" object in script 
Debug [Controller]: Connection: "[Channel1]" "beat_active" 
Debug [Controller]: Connection: "[Channel2]" "beat_active" 
Debug [Controller]: Connection: "[Channel1]" "playposition" 
Debug [Controller]: Connection: "[Channel2]" "playposition" 
Debug [Controller]: Connection: "[Master]" "crossfader" 
Debug [Controller]: Connection: "[Channel1]" "volume" 
Debug [Controller]: Connection: "[Channel2]" "volume" 
Debug [Controller]: Connection: "[Channel1]" "play" 
Debug [Controller]: Connection: "[Channel2]" "play" 
```

**Comment by markusb** [comment]
I've tested the script now also successfully in Mixxx 1.12. Works fine too.

I think it is ready to be pulled into the main tree.

File: res/controllers/MIDI_for_LightEffects.midi.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    send midi information about the musing to generate light effects with a light control app like QLCplus
+
+    This file should be placed into /usr/share/mixxx/controllers
+-->
+<MixxxMIDIPreset mixxxVersion="1.10+" schemaVersion="1">
+	<info>
+		<name>MIDI for LightEffects</name>
+		<author>DG3NEC / Michael Stahl - Markus Baertschi</author>
+		<description>MIDI-Output for light effects</description>
+		<forums>http://www.mixxx.org/forums</forums>
```
Comment (rryan) [inline res/controllers/MIDI_for_LightEffects.midi.xml]: This is usually a forum post where users can go for support. Can you make a post in the controllers sub-forum and then put the link here? That way users have a place to go to ask questions.

http://mixxx.org/forums/viewforum.php?f=7&sid=fc57cf579ddfad0bda44bc5125b819ad

Comment (rryan) [inline res/controllers/MIDI_for_LightEffects.midi.xml]: You could also add a ```<wiki>``` link here now that you have a wiki page :).

**Comment by rryan** [comment]
cool @markusb :) sorry for the delay in looking at this

**Comment by rryan** [comment]
Hi @markusb,

Any update on this? We just merged PR #824 from @dg3nec which looks pretty similar. Maybe you and @dg3nec could get together and try to merge the functionality between the two?

Cheers,
RJ

**Comment by dg3nec** [comment]
I think markus has had the same goal. Make mixxx better/easier for light control ;-) out of the box. Script functions (midi-ouput) should be the same, because it based on a old version of my "midi-for-light".

**Comment by Be-ing** [comment]
This pull request has not been updated in years so I am closing it to reduce the clutter of open PRs. I'm also not clear what the advantage of this is over the script that was merged in #824 or why we should have both of them in Mixxx.

**Pull request closed without merging**